images/stackstorm/bin/install.sh

```diff
@@ -42,11 +43,29 @@ if [ -z ${ST2MISTRAL_VERSION:-} ]; then
     ST2MISTRAL_VERSION="$(apt-cache madison st2mistral | cut -f 2 -d '|' | tr -d '[ \t]' | head -1)"
   fi
 fi
+if [ -z ${ST2CHATOPS_VERSION:-} ]; then
+  if [[ -n ${ST2_TAG:-} ]]; then
+    ST2CHATOPS_VERSION="$(apt-cache madison st2chatops | cut -f 2 -d '|' | tr -d '[ \t]' | grep ${ST2_TAG} | head -1)"
```

Comment:
This `grep` pattern is little bit error-prone...
```sh
$ ST2_TAG=3.0
$ echo 3.0.0 | grep $ST2_TAG
3.0.0
$ echo 3.3.0 | grep $ST2_TAG
3.3.0
$ echo 2.3.0 | grep $ST2_TAG
2.3.0
```
Also I think there is no need to imperatively check if the `ST2_TAG` variable has any value.
The whole block could be modified like this:
```
if [ -z ${ST2CHATOPS_VERSION:-} ]; then
  ST2CHATOPS_VERSION="$(apt-cache madison st2chatops | \
    cut -f 2 -d '|' | tr -d '[ \t]' | grep "^${ST2_TAG:-}" | head -1)"
fi
```
There might be edge cases with `.` (dot) as a reserved regex character, but that only will apply for big numbers > 100.

Reply:
I will leave edge cases with `.` (dot) for another day.

Reply:
@vutny Thanks for your constructive feedback!